### PR TITLE
Compare pitch and demo video links to ensure they are different links

### PR DESCRIPTION
**Pull request by viviancan**
Refs #3014 

This change was needed to ensure pitch and demo video links were not identical. This has been an issue in the past with team submissions. Now, a team submission cannot be saved if the video URLs are the same.

This ticket ended up being a little more involved than just adding a validation to the model. 
- `app/controllers/concerns/team_submission_video_link_review_controller.rb` I had to check if the update was successful and if so create the team activity tracker record and redirect with a success message. If the the update was not successful then check for any error messages and update the alert message accordingly

-  In `app/models/team_submission.rb` I added a custom validation method `demo_and_pitch_video_links_are_different` that compares the demo video link and the pitch video link and adds an error message. This validation only runs if the demo_video_link and pitch_video_link are present. I came to this conclusion because there are essentially 4 scenarios for the video links 
           - Both links are blank - We don't need validation here because only 1 link can be saved at a time
           - Demo link is present & pitch link blank - Validation needed
           - Demo link is blank & pitch link is present - Validation needed
           - Demo link is present and pitch link is present - Validation needed

- In `app/controllers/admin/team_submissions_controller.rb` I added a flash message if the save was unsuccessful. It was kind of confusing if both links were the same you would be redirected to the edit page without any feedback

- Finally, I added testing for the demo and pitch video links to both the mentor and student flow. I also updated a few of the previous tests in order to account for these updates. 

Hopefully, this is in a good place for code review now 🤞 You can see the updated functionality below. This represents a scenario where both the demo link and pitch link are already present, but I am trying to update my pitch link now with the same link that is used for the demo video.

https://user-images.githubusercontent.com/29210380/184239219-7db9ddb4-f802-4c1b-b946-cd14c8a187d0.mp4

### CODE REVIEW

**Review by devkonrad — APPROVED** [review]
Looks perfect to me, @viviancan! Thanks for take care of it! :)

File: app/models/team_submission.rb [hunk 1]
```diff
@@ -258,6 +258,13 @@ def developed_on?(platform_name)
       with: /[a-zA-Z0-9][a-zA-Z0-9-]{1,61}[a-zA-Z0-9]\.[a-zA-Z]{2,}(.[a-zA-Z]{2,63})?/ 
     }, 
     allow_blank: true
+
+  validate :demo_and_pitch_video_links_are_different, on: :update,
```
Comment (shaunxp20) [inline app/models/team_submission.rb:262]: 👍 I like the name you picked! 🙌 🎉 👏

File: spec/features/mentor/submission_piece_spec.rb [hunk 2]
```diff
@@ -181,6 +181,47 @@
...
+          piece: :demo_video_link
+        )
+      )
+    end
+
+    within(".pitch_video_link.incomplete") do
+      click_link "Add the pitch video link"
+    end
+
+    video_id = "XMzMyNzg3OTY1Mg"
```
Comment (shaunxp20) [inline spec/features/mentor/submission_piece_spec.rb:215]: Just a minor thing, but I had to do a double-take to make sure the `video_ids` were the same, I wonder if having only one `video_id` would be better or making this one `duplicate_video_id` or something? 🤔 😅 #JustARandomThought!

**Review by shaunxp20 — APPROVED** [review]
Looks good to me! Nice write-up explaining it all! Great improvements/catches too (regarding the admin and controller stuff)! And sweet tests! 💯 👏 🎉 🙌 :shipit:

File: spec/features/mentor/submission_piece_spec.rb (hunk 2 above)
Comment (viviancan) [reply spec/features/mentor/submission_piece_spec.rb:215]: Thanks for the suggestion 👍  I'll make that change!

**Pull request merged**